osu.Game/Rulesets/Mods/IModHasSettings.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+
+namespace osu.Game.Rulesets.Mods
+{
+    /// <summary>
+    /// An interface for mods that allows user control over it's properties.
+    /// </summary>
+    public interface IModHasSettings
+    {
+        Drawable[] CreateControls();
```

Comment:
This shouldn't pass a list of drawables, but rather, a list of settings controls maybe?

Reply:
I'd like to automatically generate it's settings based on the class' properties (maybe marked with an attribute?) but I'm not sure of that approach just yet.